file_store/src/lora_invalid_poc.rs

```diff
@@ -0,0 +1,87 @@
+use crate::{
+    datetime_from_epoch, lora_beacon_report::LoraBeaconReport,
+    lora_witness_report::LoraWitnessReport, traits::MsgDecode, Error, Result,
+};
+use chrono::{DateTime, Utc};
+use helium_proto::services::poc_lora::{
+    InvalidParticipantSide, InvalidReason, LoraBeaconReportReqV1, LoraInvalidBeaconReportV1,
+    LoraInvalidWitnessReportV1, LoraWitnessReportReqV1,
+};
+use serde::Serialize;
+
+#[derive(Serialize, Clone)]
+pub struct LoraInvalidBeaconReport {
+    pub received_timestamp: DateTime<Utc>,
+    pub reason: InvalidReason,
+    pub report: LoraBeaconReport,
+}
+
+#[derive(Serialize, Clone)]
+pub struct LoraInvalidWitnessReport {
+    pub received_timestamp: DateTime<Utc>,
+    pub reason: InvalidReason,
+    pub report: LoraWitnessReport,
+    pub participant_side: InvalidParticipantSide,
+}
+
+impl MsgDecode for LoraInvalidBeaconReport {
+    type Msg = LoraInvalidBeaconReportV1;
+}
+
+impl MsgDecode for LoraInvalidWitnessReport {
+    type Msg = LoraInvalidWitnessReportV1;
+}
+
+impl TryFrom<LoraInvalidBeaconReportV1> for LoraInvalidBeaconReport {
+    type Error = Error;
+    fn try_from(v: LoraInvalidBeaconReportV1) -> Result<Self> {
+        let invalid_reason: InvalidReason = InvalidReason::from_i32(v.reason)
+            .ok_or_else(|| Error::custom("unsupported invalid_reason"))?;
+        Ok(Self {
+            received_timestamp: datetime_from_epoch(v.received_timestamp),
+            reason: invalid_reason,
+            report: LoraBeaconReport::try_from(v.report.unwrap())?,
+        })
+    }
+}
+
+impl TryFrom<LoraInvalidBeaconReport> for LoraInvalidBeaconReportV1 {
```

Comment:
do we need this? doesn't the type safety assume a well-formed `LoraInvalidBeaconReport` struct such that you only need to encode it to get a `LoraInvalidBeaconReportV1` or call `into` to easily convert it? i understand the need for the `TryFrom` the proto struct into the internal struct but i would think this wouldn't be necessary

Reply:
yeah good point.  to hasty with the copy pasta

Reply:
fixed here: c0be07d313edd828dc9308501946c2cfe445b779